internal/gps/constraints.go

```diff
@@ -52,7 +52,7 @@ type Constraint interface {
 // If the input string cannot be made into a valid semver Constraint, an error
 // is returned.
 func NewSemverConstraint(body string) (Constraint, error) {
```

Comment:
I think we actually might want to mirror the change in `Masterminds/semver` instead, and introduce a `func NewSemverConstraintIC()`. That way we still give the implementing tool the choice as to whether they're using implicit carets or not; gps can remain agnostic.